scripts/test/diff_dashboard_states.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+#
+# Requirements:
+# - Imagemagick installed with PNG support locally
+
+BRANCH1=$(git rev-parse --abbrev-ref HEAD)
+BRANCH2=$1
+
+shift 1
+
+# verify the branch exists
+if ! git rev-parse --verify $BRANCH2 > /dev/null
+then
+  echo "$BRANCH2 is not a valid branch"
+  exit 1
+fi
+
+if ! type "magick" > /dev/null
+then
+  echo "ImageMagick is not installed"
+  exit 1
+fi
+
+set -e -o pipefail
+
+OUT1="output/dashboard_states_current"
+OUT2="output/dashboard_states_other"
+DIFFOUT="output/dashboard_states_diff"
+
+mkdir -p output
+chmod o+rwx output
+
+./scripts/test/run_snapshot_dashboard_states.sh --output "$OUT1" $@
+
+git fetch
+git checkout "$BRANCH2"
+
+./scripts/test/run_snapshot_dashboard_states.sh --output "$OUT2" $@
+
+git checkout "$BRANCH1"
+
+mkdir -p "$DIFFOUT"
+
+set +e # imagemagick returns nonzero codes, so ignore them
+
+for f1 in $OUT1/*.png ; do
+  fname=$(basename $f1)
+  echo "Diffing: $fname"
+  magick compare "$OUT1/$fname" "$OUT2/$fname" "$DIFFOUT/$fname"
+done
```

Comment:
Can you add an `exit 0` at the end so the script doesn't return the non-zero exit code for `magick`?